docs/gitbook/tutorials/alibabacloud-servicemesh-progressive-delivery.md

```diff
@@ -0,0 +1,171 @@
+# Alibaba Cloud Service Mesh Canary Deployments
+
+This guide shows you how to use Alibaba Cloud Service Mesh(ASM) and Flagger to automate canary deployments.
+You'll need an ACK(Alibabacloud Container Service for Kubernetes) cluster (Kubernetes >= 1.16) configured with ASM(Alibaba Cloud Service Mesh), you can find the installation guide [here](https://docs.flagger.app/install/flagger-install-on-alibabacloud-servicemesh).
+
+## Prerequisites
+
+- Created an ACK([Alibabacloud Container Service for Kubernetes](https://cs.console.aliyun.com)) cluster instance.
+- Created an ASM([Alibaba Cloud Service Mesh](https://servicemesh.console.aliyun.com)) instance, and added ACK cluster.
+
+### Variables declaration
+
+- `$ACK_CONFIG`: the kubeconfig file path of ACK, which be treated as`$HOME/.kube/config` in the rest of guide.
+- `$MESH_CONFIG`: the kubeconfig file path of ASM.
+- `$ISTIO_RELEASE`: see https://github.com/istio/istio/releases
+- `$FLAGGER_SRC`: see https://github.com/fluxcd/flagger
+
+## Bootstrap
+
+### Setup EnvoyFilters for Metrics Monitoring
+To automate canary deployments, Flagger uses the metrics from Prometheus, which are generated from Data Plane of Service Mesh, and configured by EnvoyFilters.
+
+These EnvoyFilters could be generated automatically, if we choose one of the below options :
+- Enable `Telemetry` using [UpdateMeshFeature](https://www.alibabacloud.com/help/doc-detail/171592.htm) API
+- Enable Metrics Monitoring from [ASM Console](https://servicemesh.console.aliyun.com/)
+
+### Setup Gateway
```

Comment:
Isn't this a copy/paste from Istio tutorial? I don't see anything specific to ASM, people should follow the Istio tutorials for canary and a/b testing. Also the `Variables declaration` and `Setup EnvoyFilters` should be in the install docs instead of here. I propose that in thr ASM install docs, you like to the 2 Istio tutorials instead of this.

Reply:
done